src/components/WorkOrders/CloseWorkOrderFormComponents/FollowOnRequestDifferentTradesForm.js

```diff
@@ -1,11 +1,61 @@
 import { FOLLOW_ON_REQUEST_AVAILABLE_TRADES } from '@/utils/statusCodes'
-import { Checkbox } from '../../Form'
+import { Checkbox, DataList } from '../../Form'
+import { useState, useEffect } from 'react'
+import ErrorMessage from '../../Errors/ErrorMessage'
+
+import { getTrades } from '@/root/src/utils/requests/trades'
 
 const FollowOnRequestDifferentTradesForm = (props) => {
-  const { register, requiredFollowOnTrades } = props
+  const { register, requiredFollowOnTrades, watch, errors } = props
+
+  const [trades, setTrades] = useState([])
+  const [filteredTrades, setFilteredTrades] = useState([])
+  const [error, setError] = useState(null)
+  const maxLength = 100
+  const [remainingCharacterCount, setRemainingCharacterCount] = useState(100)
 
   const selectedTrades = new Set(requiredFollowOnTrades.map((x) => x.name))
 
+  const isDifferentTradesChecked = watch('followon-trades-other')
+
+  useEffect(() => {
+    fetchTrades()
+  }, [])
+
+  useEffect(() => {
+    filterTrades()
+  }, [trades])
+
+  const fetchTrades = async () => {
+    const tradesResponse = await getTrades()
+    if (!tradesResponse.success) {
+      setError(tradesResponse.error)
+      return
+    }
+    setTrades(tradesResponse.response)
+  }
+
+  const filterTrades = () => {
+    const codesToFilter = [
+      'CP',
+      'DR',
+      'GR',
+      'GS',
+      'EL',
+      'MU',
+      'PN',
+      'PL',
+      'RF',
+      'SC',
+      'UP',
+      'SV',
+    ]
+    const filteredTrades = trades.filter(
+      (trade) => !codesToFilter.includes(trade.key)
+    )
```

Comment:
See if you can convert this to a `Set<string>`. It would be a lol cleaner. Secondly, this codes are a constant, so I would remove them from the render function (above the component code).